proto/axelar/nexus/exported/v1beta1/types.proto

```diff
@@ -87,3 +87,30 @@ enum TransferDirection {
   TRANSFER_DIRECTION_OUTGOING = 2
       [ (gogoproto.enumvalue_customname) = "Outgoing" ];
 }
+
+message GeneralMessage {
+  enum Status {
+    option (gogoproto.goproto_enum_prefix) = false;
+    option (gogoproto.goproto_enum_stringer) = true;
+
+    STATUS_UNSPECIFIED = 0 [ (gogoproto.enumvalue_customname) = "NonExistent" ];
+    STATUS_APPROVED = 1 [ (gogoproto.enumvalue_customname) = "Approved" ];
+    STATUS_EXECUTED = 2 [ (gogoproto.enumvalue_customname) = "Executed" ];
+  }
+
+  string id = 1 [ (gogoproto.customname) = "ID" ];
+  string source_chain = 2 [ (gogoproto.casttype) = "ChainName" ];
+  string sender = 3;
+  string destination_chain = 4 [ (gogoproto.casttype) = "ChainName" ];
+  string receiver = 5;
+  bytes payload_hash = 6;
+  Status status = 7;
+  oneof message {
+    PureMessage pure_message = 8;
+    MessageWithToken message_with_token = 9;
+  }
+}
```

Comment:
why does this need to be a oneof, instead of allowing the token to be nil?

Reply:
I prefer that too